### ISSUE TITLE
[Bug] using $parent in sub-select in where clause result in NPE

### ISSUE THREAD

**Issue by matanshukry** [orientechnologies/orientdb#9242]
## Expected behavior
Not an NPE

## Actual behavior  
NPE  

## Steps to reproduce  
Make a query with an inner sub select and use $parent in it.

For example:
```
select from user where userId = (select from user where userId != $parent.userId)
```

The NPE comes from this:
https://github.com/orientechnologies/orientdb/blob/3.1.0-M3/core/src/main/java/com/orientechnologies/orient/core/sql/executor/OSelectExecutionPlanner.java#L1454

This is null: `this.statement.getLetClause()`
That is, the let clause of the statement is null (since there is no actual let). There is `info.perRecordLetClause` since the code made the sub select a record, and it uses $parent which makes it a per record let clause (rather than a global clause).

**Comment by tglman** [comment]
Hi @matanshukry,

The 3.1.0 CE was released last week, could you try to reproduce the case with that release ?

Regards

**Comment by matanshukry** [comment]
@tglman yeah it's solved:
https://github.com/orientechnologies/orientdb/commit/90ece2331fc342137e88294ba0d1c548f5748872

Look like the commit that fixed it was a while ago btw; weird it isn't 3.1.0-M3.
